File: Medium/73_set_matrix_zeroes/solution.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        vertical = set()
        horizontal = set()
        for m in range(len(matrix)):
            for n in range(len(matrix[0])):
                if matrix[m][n] == 0:
                    vertical.add(n)
                    horizontal.add(m)

        for v in vertical:
            for m in range(len(matrix)):
                matrix[m][v] = 0

        for h in horizontal:
            for n in range(len(matrix[0])):
                matrix[h][n] = 0

        

File: Medium/73_set_matrix_zeroes/solution.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        std::vector<int> vertical;
        std::vector<int> horizontal;
        for (int m=0; m<matrix.size(); m++) {
            for (int n=0; n<matrix[0].size(); n++) {
                if (matrix[m][n] == 0) {
                    vertical.push_back(n);
                    horizontal.push_back(m);
                }
            }
        }

        for (int v : vertical) {
            for (int i=0; i<matrix.size(); i++) {
                matrix[i][v] = 0;
            }
        }

        for (int h : horizontal) {
            for (int i=0; i<matrix[0].size(); i++) {
                matrix[h][i] = 0;
            }
        }
    }
};
